### PR TITLE
[terra-dev-site v7] New Props Table

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary

Update props table to only parse MDX once, and better display content.

Remove references to infinite list.

Closes #

### Deployment Link

See the new props table here:
https://terra-dev-si-new-props--swiv7m.herokuapp.com/dev_tools/cerner-terra-dev-site/terra-dev-site/webpack-loaders/props-table-loader

Compare to: https://engineering.cerner.com/terra-ui/dev_tools/terra-dev-site/terra-dev-site/webpack-loaders/props-table-loader

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: src/webpack/loaders/devSitePropsTable.js [hunk 1]
```diff
@@ -1,59 +1,93 @@
...
-const mdx = require('@mdx-js/mdx');
 const reactDocs = require('react-docgen');
 const findSource = require('../loaderUtils/findSource');
 
 /**
- * Create an inline mdx react component.
- * @param {*} description
- * @param {*} callback
+ * Parses prop types into a more compact representation.
+ * @param {object} type prop type to parse.
```
Comment (ryanthemanuel) [inline src/webpack/loaders/devSitePropsTable.js:7]: I like to call out the specific elements in the object if possible: 

```suggestion
 * @param {object} type prop type information to parse.
 * @param {string} type.value the value for the type
 * @param {string} type.name the name of the type
 * @param {boolean} type.required whether or not the type is required
 * @param {description} type.description the description of the type
```

**Review by ryanthemanuel — APPROVED** [review]

File: src/loader-components/PropsTable.module.scss [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
+    font-weight: 600;
   }
 
-  .props-tr {
-    vertical-align: top;
+  .table {
+    border-collapse: collapse;
+    border-spacing: 0;
+    display: block;
+    margin-bottom: 16px;
```
Comment (tbiethman) [inline src/loader-components/PropsTable.module.scss:13]: You could use rems here if you're worried about adding font scaling to dev-site in the future.

File: src/webpack/loaders/devSitePropsTable.js [hunk 3]
```diff
@@ -1,59 +1,97 @@
...
+    } else if (name === 'shape') {
+      returnValue = [
+        'shape: {',
+        ...Object.entries(value).reduce((acc, [key, val]) => acc.concat([`${key}: {`, ...parsePropType({ type: val, indent: true }), '},']), []).map((val) => `  ${val}`),
+        '},',
+      ];
+    } else if (name === 'exact') {
+      returnValue = [
+        'exactShape: {',
+        ...Object.entries(value).reduce((acc, [key, val]) => acc.concat([`${key}: {`, ...parsePropType({ type: val, indent: true }), '},']), []).map((val) => `  ${val}`),
```
Comment (tbiethman) [inline src/webpack/loaders/devSitePropsTable.js:61]: I trust you

File: src/loader-components/_PropsTable.jsx [hunk 4]
```diff
@@ -3,80 +3,105 @@ import React, { useContext } from 'react';
...
+  /**
+   * Is the cell required?
+   */
+  isRequired: PropTypes.bool,
+};
+
+const PropNameCell = ({ children }) => (
+  <Cell
+    className={cx('bold')}
+    key="PROP"
```
Comment (tbiethman) [inline src/loader-components/_PropsTable.jsx:33]: I don't think these keys are necessary, but they don't hurt anything either if you like having them.

Comment (mjhenkes) [reply src/loader-components/_PropsTable.jsx:33]: I added them based on the html table guides. 🤷‍♂️

Comment (tbiethman) [reply src/loader-components/_PropsTable.jsx:33]: The guide might be mapping over a data set and building an array of Cells, in which case the keys would be necessary.

**Review by tbiethman — APPROVED** [review]

File: src/loader-components/_PropsTable.jsx (hunk 4 above)
Comment (mjhenkes) [reply src/loader-components/_PropsTable.jsx]: Removed

File: src/loader-components/PropsTable.module.scss (hunk 2 above)
Comment (mjhenkes) [reply src/loader-components/PropsTable.module.scss]: updated

**Pull request merged**